components/layout_thread_2020/lib.rs

```diff
@@ -1106,34 +1112,42 @@ impl LayoutThread {
         }
     }
 
-    /// Update layout given a new viewport. Returns true if the viewport changed or false if it didn't.
-    fn handle_viewport_change(
-        &mut self,
-        window_size_data: WindowSizeData,
-        guards: &StylesheetGuards,
-    ) -> bool {
-        // If the viewport size and device pixel ratio has not changed, do not make any changes.
-        let au_viewport_size = Size2D::new(
+    fn viewport_did_change(&mut self, window_size_data: WindowSizeData) -> bool {
+        let new_pixel_ratio = window_size_data.device_pixel_ratio.get();
+        let new_viewport_size = Size2D::new(
             Au::from_f32_px(window_size_data.initial_viewport.width),
             Au::from_f32_px(window_size_data.initial_viewport.height),
         );
 
-        if self.stylist.device().au_viewport_size() == au_viewport_size &&
-            self.stylist.device().device_pixel_ratio().get() ==
-                window_size_data.device_pixel_ratio.get()
-        {
-            return false;
-        }
+        // TODO: eliminate self.viewport_size in favour of using self.device.au_viewport_size()
+        self.viewport_size = new_viewport_size;
```

Comment:
@nicoburns is it ok to set the `self.viewport` size even if it hasn't been changed?

Reply:
Yes. This is a plain struct, so that will have no observable effect.